### PR TITLE
Remove jfrog-client-go dependency

### PR DESCRIPTION
**Pull request by alexhung**
- Copy the structs from jfrog-client-go for compatibility. No attempt is made to improve the current code.
- Add deprecation message for `artifactory_permission_targets` which has not been part of official documentation for 2.3.1

### CODE REVIEW

**Review by maheshjfrog — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## 6.3.0 (Apr 18, 2022)
+
+IMPROVEMENTS:
+
+* resource/artifactory_permission_targets: Add deprecation message [GH-413]
+* Removed dependency on `jfrog-client-go` package [GH-413]
+
+NOTES:
+
+* Resource `artifactory_permission_targets` is deprecated and will be removed in next major release. Resource `artifactory_permission_target` (singular) has identical schema which will allow straightforward migration.
```
Comment (chb0github) [inline CHANGELOG.md:10]: This deprecation notice is obviously not right. Perhaps changing it to -in a future major release"

**Review by chb0github — CHANGES_REQUESTED** [review]
Change deprecation notice

File: CHANGELOG.md (hunk 1 above)
Comment (alexhung) [reply CHANGELOG.md:10]: Why is this not right? Next major release will be v7 which we haven't even planned yet.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,14 @@
+## 6.3.0 (Apr 18, 2022). Tested on Artifactory 7.37.13
+
+IMPROVEMENTS:
+
+* resource/artifactory_permission_targets: Add deprecation message [GH-413]
+* Removed dependency on `jfrog-client-go` package [GH-413]
+
+NOTES:
+
```
Comment (chb0github) [inline CHANGELOG.md:9]: In the...

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+* Resource `artifactory_permission_targets` is deprecated and will be removed in next major release. Resource `artifactory_permission_target` (singular) has identical schema which will allow straightforward migration.
```
Comment (chb0github) [inline CHANGELOG.md:10]: Has an identical schema to...

I had to read this more than once

**Review by chb0github — APPROVED** [review]
Tiny grammar and sentence changes. I leave it to you to decide

**Review by chb0github — APPROVED** [review]

**Pull request merged**